src/pypeh/core/models/validation_dto.py

```diff
@@ -395,143 +396,103 @@ def get_dataset_validations(
     def get_dataset_identifier_consistency_validations_dict(
         cls,
         observation_list: Sequence[peh.Observation],
-        layout: peh.DataLayout,
-        dataset_mapping: Dict[str, Dict[str, str | int | Dict[str, Sequence[str]]]],
+        data_import_config: peh.DataImportConfig,
         data_dict: Dict[str, Dict[str, Sequence] | T_DataType],
+        cache: CacheContainerView,
     ) -> Dict[str, Sequence[ValidationDesign]] | None:
         """Returns validation designs that verify consistency of the entity identifiers in the data."""
 
-        observation_dict = {o.id: o for o in observation_list}
-        observation_id_to_dataset_label_dict = {}
-        identifying_observation_list = []
-        for dataset_label, mapping in dataset_mapping.items():
-            if "observation_id" in mapping and mapping["observation_id"] is not None:
-                observation_id = mapping["observation_id"]
-                observation_id_to_dataset_label_dict[observation_id] = dataset_label
-                if str(observation_dict[observation_id].observation_type) == "metadata":
-                    identifying_observation_list.append(observation_dict[observation_id])
-
-        entity_type_identifiers_dict = {}
-        for observation in identifying_observation_list:
-            observable_entity_type = str(observation.observation_design.observable_entity_type)
-            if observable_entity_type in entity_type_identifiers_dict.keys():
-                raise AttributeError(
-                    f"Found multiple competing metadata observations for the {observable_entity_type} EntityType"
-                )
-            else:
-                entity_type_identifiers_dict[observable_entity_type] = {}
-                for prop in observation.observation_design.identifying_observable_property_id_list:
-                    entity_type_identifiers_dict[observable_entity_type][prop] = list(
-                        data_dict[observation_id_to_dataset_label_dict[observation.id]][prop]
-                    )
+        def get_identifier_values(section_id, prop_label, data_import_config, data_dict):
+            identifier_values = None
+            observation_id_list = []
+            for link in data_import_config.section_mapping.section_mapping_links:
```

Comment:
This is built from the `data_import_config`. The  `DataImportConfig` should not go beyond the data import. This seems to suggest that we'd want to build these validation rules upon importing the data.